docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,416 @@
+
+<xref:System.Text.Json.Utf8JsonWriter?displayProperty=fullName> is a high-performance way to write UTF-8 encoded JSON text from common .NET types like `String`, `Int32`, and `DateTime`. The writer is a low-level type that can be used to build custom serializers.
+
+The following sections explain recommended programming patterns for using `Utf8JsonReader` and `Utf8JsonWriter`.
+
+### Ref structs
+
+The `Utf8JsonReader` and `Utf8JsonWriter` types are *ref structs*, which means they have certain limitations. For example, they can't be stored as a field on a class or struct other than a ref struct. To achieve high performance, these types must be `ref structs` since they need to cache the input or output `Span<byte>`, which itself is a ref struct. In addition, these types are mutable since they hold state. Therefore, you should **pass them by ref** rather than by value. Passing them by value would result in a struct copy and the state changes would not be visible to the caller. This differs from `Newtonsoft.Json` since the `Newtonsoft.Json` `JsonTextReader` and `JsonTextWriter` are classes. For more information about how to use ref structs, see [Write safe and efficient C# code](../../csharp/write-safe-efficient-code.md).
+
+### Read with a Stream or PipeReader
+
+The `Utf8JsonReader` supports reading from a UTF-8 encoded `ReadOnlySpan<byte>` or `ReadOnlySequence<byte>` (which is the result of reading from a `PipeReader`). For synchronous reading, you could read the JSON payload until the end of the stream into a byte array and pass that into the reader. For reading from a string (which is encoded as UTF-16), you should use the `Encoding.UTF8.GetBytes` API to first transcode the string to a UTF-8 encoded byte array. Then pass that to the `Utf8JsonReader`. For code examples, see [Use Utf8JsonReader](system-text-json-how-to.md#use-utf8jsonreader).
+
+### Read with multi-segment ReadOnlySequence
+
+If your JSON input is a `ReadOnlySpan<byte>`, each JSON element can be accessed from the `ValueSpan` property on the reader as you go through the read loop. However, if your input is a `ReadOnlySequence<byte>` (which is the result of reading from a `PipeReader`), some JSON elements might straddle multiple segments of the `ReadOnlySequence<byte>` object. These elements would not be accessible from `ValueSpan` in a contiguous memory block. Instead, whenever you have a multi-segment `ReadOnlySequence<byte>` as input, you should always poll the `HasValueSequence` property on the reader to figure out how to access the current JSON element. Here's a recommended pattern:
+
+```csharp
+while (reader.Read())
+{
+  switch (reader.TokenType)
+  {
+    // ...
+    ReadOnlySpan<byte> jsonElement = reader.HasValueSequence ?
+                reader.ValueSequence.ToArray() :
+                reader.ValueSpan;
+    // ...
+  }
+}
+```
+
+### Read and write with UTF-8 text
```

Comment:
We should provide a workaround for `writer.WriteRawValue`
https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonWriter_WriteRaw.htm
https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonWriter_WriteRawValue.htm

One workaround is:
```C#
using JsonDocument doc = JsonDocument.Parse(string);
doc.WriteTo(writer);
```